### ISSUE TITLE
v7.9.2 seems to introduce CORS issue with Firebase Storage

### ISSUE THREAD

**Issue by lupas** [firebase/firebase-js-sdk#2695]
### Describe your environment

  * Operating System version: Mac OS Catalina 10.15.3 (19D76)
  * Browser version: 80.0.3987.122 (Official Build) (64-bit)
  * Firebase SDK version: 7.9.2
  * Firebase Product: Storage

### Describe the problem
With 7.9.2, Firebase Storage .put() throws the following CORS error on localhost:

> Access to fetch at 'https://firebasestorage.googleapis.com/v0/b/appId.appspot.com/o?name=_userdata%2FYixPa2SRSiRZ6OaYrDJN9cFLL6t1%2FuserPic.jpg'
from origin 'http://localhost:3000' has been blocked by CORS policy:
Request header field x-firebase-gmpid is not allowed by Access-Control-Allow-Headers
in preflight response.

Also see StackOverflow issue [here](https://stackoverflow.com/questions/60456895/firebase-request-header-field-x-firebase-gmpid-is-not-allowed-by-access-contro).

Same code works with 7.9.1.

#### Steps to reproduce:
Seems to appear anytime `put()`, `putString()` and so on is called to upload a file to Firebase Storage.

#### Relevant Code:

Example:

```javascript
async upload(base64Img, metadata, path) {
  const storageRef = storage.
    .ref()
    .child(`${path}/photo.jpg`)

  try {
    await storageRef.putString(base64Img, 'base64', metadata)
  } catch (e) {
    // handle error
  }
}
```

**Comment by SarhadSalam** [comment]
I was facing the same issue too, I am almost confident the error is in 7.9.2 version of the SDK. With 7.9.0 (the version I am using), my resources are fetched properly.

**Comment by lupas** [comment]
> I was facing the same issue too, I am almost confident the error is in 7.9.2 version of the SDK. With 7.9.0 (the version I am using), my resources are fetched properly.

Yes agreed, error did not appear in v7.9.1, v7.9.0 and before.

**Comment by schmidt-sebastian** [comment]
The next release should unblock these requests again. For now, please stay with 7.9.1.

**Comment by schmidt-sebastian** [comment]
Fixed in 7.9.3.
